### PR TITLE
Add namespaces and proper support of the pointers

### LINKED ISSUE #7049: Support of namespaces in the gen_java.py
Right now modules with dedicated namespace are not supported. I've made a changes suggested in https://github.com/opencv/opencv/issues/6605 to enable this support.

### PR DESCRIPTION
**Pull request by smigielski**
resolves #7049
### This pullrequest changes

Add support for namespaces in gen_java.py

### CODE REVIEW

**Comment by mshabunin** [comment]
@smigielski , please fix the issue:
```
  File "/build/precommit_linux64/opencv/modules/java/generator/gen_java.py", line 1541
    return False
               ^
TabError: inconsistent use of tabs and spaces in indentation
```

**Comment by mshabunin** [comment]
```
modules/java/generator/gen_java.py:1445: tab in indent.
+	namespace = ('using namespace ' + ci.namespace.replace('.', '::') + ';') if ci.namespace else ''
modules/java/generator/gen_java.py:1541: tab in indent.
+	return False
```

File: modules/java/generator/gen_java.py [hunk 1]
```diff
@@ -1535,7 +1538,7 @@ def isSmartClass(self, classname):
         '''
         Check if class stores Ptr<T>* instead of T* in nativeObj field
         '''
-        return self.isWrapped(classname) and self.classes[classname].base
+        return False
```
Comment (mshabunin) [inline modules/java/generator/gen_java.py]: Why this change was made?

Comment (mshabunin) [inline modules/java/generator/gen_java.py]: @smigielski , ping.

Comment (smigielski) [inline modules/java/generator/gen_java.py]: Sorry I was on vacations. This method was causing that objects are passing to java as Ptr<T>  whereas in java it was casted to T directly, which was not working at all. 

BTW. It seems that there is a memory leak issue there so some further work is needed there. 

https://github.com/opencv/opencv/pull/6815/files/ef450050561b9107f3cb7eff57e27917dc4165dc#r76603254

I will try to think about it and change it later this week.

Comment (mshabunin) [inline modules/java/generator/gen_java.py]: Maybe it will be better to revert line 1350 from previous PR and line 1541 in this PR (probably `self.classes[classname].base` check should be omitted).
